### PR TITLE
feat(tab-group): add standalone tab navigation primitives

### LINKED ISSUE #272: Add TabGroup component
Add a composable TabGroup package for building synchronized tab navigation and swipeable content in ReactLynx applications.

The public API should include:

- `TabsRoot`
- `TabsBar`
- `TabsItem`
- `TabsIndicator`
- `TabsPanel`

`TabsPanel` should integrate with `ViewPager` for synchronized swipe navigation. The package should be exported from `@lynx-js/lynx-ui` and include API documentation and runnable examples.

Release setup:

- Bootstrap the `@lynx-js/lynx-ui-tab-group` and `@lynx-example/lynx-ui-tab-group` npm package names.
- Configure trusted publishing for both packages.

### PR DESCRIPTION
**Pull request by f0rdream**
Adds composable tab navigation with `TabsRoot`, `TabsBar`, `TabsItem`, and `TabsIndicator`, including animated indicators and runnable examples.

Targets `main` independently of ViewPager. Swipeable `TabsPanel` integration is handled in #290.

Part of #272.

- Introduce composable `TabsRoot`, `TabsBar`, `TabsItem`, and `TabsIndicator` primitives with animated indicators.
- Export TabGroup APIs through `@lynx-js/lynx-ui`.
- Bootstrap TabGroup packages, documentation, skills, and runnable examples.
- Preserve generic `TabsData<T>` signatures and inherited `TabItemProps` in API documentation.
- Refine selection callbacks, smooth selection, initial tab scrolling, and caller event and ref handling.
- Align `BaseScrollEvents` with dedicated Lynx event types.
- Cover indicator geometry and TypeDoc rendering edge cases with Vitest tests.

### CODE REVIEW

File: apps/examples/TabGroup/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@lynx-example/lynx-ui-tab-group",
+  "version": "0.0.1",
```
Comment (Dugyu) [inline apps/examples/TabGroup/package.json]: "version" should starts from "0.0.16" (or  higher, depends on the date when this PR is merged), aligning with other lynx-ui examples in the repo.

Comment (f0rdream) [reply apps/examples/TabGroup/package.json]: Updated

**Comment by Dugyu** [comment]
## PR #273 documentation issues

There are two API documentation issues to address:

1. **Preserve the `TabsData<T>` generic parameter**

   `packages/lynx-ui-tab-group/src/types/index.docs.ts` correctly declares:

   - `TabsData<T>` at line 140
   - `data: TabsData<T>[]` at line 176
   - `renderTabItem?: (tabItem: TabsData<T>) => ReactNode` at line 192

   However, `packages/lynx-ui-tab-group/docs/APIReference.mdx` renders these as `TabsData[]` and `TabsData`, losing the generic argument. Please update the TypeDoc rendering logic to preserve type arguments and regenerate `APIReference.mdx`.

2. **Do not remove supported inherited properties from `TabItemProps`**

   `tools/typedoc/tpl-data.ts:580` filters all inherited properties specifically for `TabItemProps`. As a result, the generated `TabItem` API table only contains `tabKey`.

   However, `packages/lynx-ui-tab-group/src/TabsItem.tsx:24` also consumes `children`, `className`, and `style`. These supported public properties should remain visible in the API reference. Please narrow or remove this special case and regenerate `APIReference.mdx`.

As a non-blocking documentation cleanup, the complete usage examples in `packages/lynx-ui-tab-group/README.md` and `README_zh.md` are more detailed than the repository's package README guidance. Consider keeping these files focused on purpose, installation, component structure, and links to the full examples under `apps/examples/TabGroup`.

File: apps/examples/TabGroup/CHANGELOG.md
Comment (Dugyu) [inline apps/examples/TabGroup/CHANGELOG.md:1]: Let's remove this CHANGELOG.md

File: apps/examples/TabGroup/package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@lynx-example/lynx-ui-tab-group",
+  "version": "0.0.16",
```
Comment (Dugyu) [inline apps/examples/TabGroup/package.json]: 0.0.17

**Review by Dugyu — CHANGES_REQUESTED** [review]
CHANGELOG.md must be removed before merging.

**Review by MoonfaceX — DISMISSED** [review]

File: apps/examples/TabGroup/LICENSE
Comment (Dugyu) [inline apps/examples/TabGroup/LICENSE:1]: Replace this file with a standard Apache License.

File: apps/examples/TabGroup/lynx.config.mjs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+import { exampleConfig } from '../../../tools/configs/exampleConfig.mjs'
+
+const defaultConfig = exampleConfig(
+  {
+    PrimitivesTabsBasic: './Basic/index.tsx',
+    PrimitivesTabsMotion: './Motion/index.tsx',
+  },
```
Comment (Dugyu) [inline apps/examples/TabGroup/lynx.config.mjs:16]: Remove `Primitives-*` prefix.
Change `Tabs-*` to `TabGroup-*`

File: packages/lynx-ui-tab-group/package.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    "url": "git+https://github.com/lynx-family/lynx-ui.git",
+    "directory": "packages/lynx-ui-tab-group"
+  },
+  "license": "Apache-2.0",
+  "author": {
+    "name": "The Lynx Authors"
+  },
+  "sideEffects": [
+    "**/*.css"
+  ],
```
Comment (Dugyu) [inline packages/lynx-ui-tab-group/package.json]: Use `sideEffects: false` ?

**Review by Dugyu — CHANGES_REQUESTED** [review]

File: apps/examples/TabGroup/Basic/index.css [hunk 5]
```diff
@@ -0,0 +1,46 @@
+@import "@lynx-js/luna-styles/index.css";
+
+.tab-group-demo-basic {
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  padding-top: calc(48px + env(safe-area-inset-top));
```
Comment (Dugyu) [inline apps/examples/TabGroup/Basic/index.css]: Prefer not to use `env(safe-area-inset-top)` here, as this API is currently discouraged and will likely be replaced by a different implementation in the future.

File: apps/examples/TabGroup/Motion/index.css [hunk 6]
```diff
@@ -0,0 +1,46 @@
+@import "@lynx-js/luna-styles/index.css";
+
+.tab-group-demo-motion {
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  padding-top: calc(48px + env(safe-area-inset-top));
```
Comment (Dugyu) [inline apps/examples/TabGroup/Motion/index.css]: Same here, please avoid using `env(safe-area-inset-top)` .

File: apps/examples/TabGroup/Basic/index.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  const tabsArray = ['Home', 'Discover', 'Messages', 'Profile']
+  const [tabs] = useState<TabsData<string>[]>(
+    Array.from({ length: tabsArray.length }, (_, index) => ({
+      tabItem: tabsArray[index],
+      getTabKey: () => tabsArray[index],
+    })),
+  )
+
+  return (
+    <view className='tab-group-demo-basic'>
```
Comment (Dugyu) [inline apps/examples/TabGroup/Basic/index.tsx]: Add `lunaris-dark`

File: apps/examples/TabGroup/Basic/index.css [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  gap: 24px;
```
Comment (Dugyu) [inline apps/examples/TabGroup/Basic/index.css:9]: Add `  background-color: var(--canvas);`

File: apps/examples/TabGroup/Basic/index.css [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+.tab-group-demo-basic__tab-item {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: max-content;
+  padding-left: 40px;
+  padding-right: 40px;
```
Comment (Dugyu) [inline apps/examples/TabGroup/Basic/index.css:31]: Add `color: var(--content);`

**Review by Dugyu — CHANGES_REQUESTED** [review]

File: apps/examples/TabGroup/Motion/index.css [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  gap: 24px;
```
Comment (Dugyu) [inline apps/examples/TabGroup/Motion/index.css:9]: Add `background-color: var(--canvas);`

File: apps/examples/TabGroup/Motion/index.css [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+.tab-group-demo-motion__tab-item {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: max-content;
+  padding-left: 40px;
+  padding-right: 40px;
```
Comment (Dugyu) [inline apps/examples/TabGroup/Motion/index.css:31]: Add `color: var(--content);`

File: apps/examples/TabGroup/Motion/index.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  const tabsArray = ['Spring', 'tuned', 'for', 'indicator', 'motion', 'config']
+  const [tabs] = useState<TabsData<string>[]>(
+    Array.from({ length: tabsArray.length }, (_, index) => ({
+      tabItem: tabsArray[index],
+      getTabKey: () => tabsArray[index],
+    })),
+  )
+
+  return (
+    <view className='tab-group-demo-motion'>
```
Comment (Dugyu) [inline apps/examples/TabGroup/Motion/index.tsx]: Add `lunaris-dark` class

**Review by Dugyu — CHANGES_REQUESTED** [review]

File: apps/examples/TabGroup/package.json [hunk 13]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@lynx-example/lynx-ui-tab-group",
+  "version": "0.0.17",
```
Comment (Dugyu) [inline apps/examples/TabGroup/package.json]: 0.0.18

File: packages/lynx-ui-common/src/types/interfaces/BaseScrollEvents.ts
Comment (Dugyu) [inline packages/lynx-ui-common/src/types/interfaces/BaseScrollEvents.ts:1]: This change fixes the scroll event parameter types but affects several downstream packages. Related API documentation and changesets should be updated to ensure the corrected types are published consistently.

**Review by Dugyu — DISMISSED** [review]
This PR updates public scroll event types inherited by several packages. Please add patch changesets for:

- `@lynx-js/lynx-ui-common`
- `@lynx-js/lynx-ui-list`
- `@lynx-js/lynx-ui-scroll-view`
- `@lynx-js/lynx-ui-feed-list`
- `@lynx-js/lynx-ui-input`

`@lynx-js/lynx-ui-tab-group` and `@lynx-js/lynx-ui` are already covered by the existing changesets. Please also regenerate the affected API reference files.

**Comment by Dugyu** [comment]
## Review summary

Reviewed the latest PR head (`4c21d50`) with the following findings: 

1. **Selection behavior incorrectly depends on `TabsIndicator`**
   (`TabsIndicator.tsx:123-163`)

   `onTabChanged` is dispatched exclusively from `TabsIndicator`.
   Omitting the indicator prevents the callback from firing, while
   rendering multiple indicators causes duplicate callbacks.

   Imperative selection is also processed by both `selectTarget` and
   `tabSelectIndex`. The second listener can override the per-call
   `smooth` argument with the root-level `selectBehavior`.

2. **Union array types are rendered with incorrect precedence**
   (`tools/typedoc/tpl-data.ts:390-393`)

   The current renderer converts `Array<string | number>` into
   `string | number[]` instead of `(string | number)[]`. Regenerating
   documentation reproduces this incorrect contract in existing APIs
   such as `Sheet.snapPoints`.

3. **`TabsItem` silently overrides inherited event and ref props**
   (`TabsItem.tsx:95-102`)

   `TabItemProps` extends `StandardProps`, which exposes `bindtap`,
   `main-thread:bindlayoutchange`, and `main-thread:ref`. Because the
   implementation spreads caller props first and then supplies its own
   values, these caller-provided props never take effect.

   Please omit internally owned keys from the public type or explicitly
   compose supported caller handlers.

4. **The initial selected tab is not scrolled into view**
   (`TabsItem.tsx:57-66`)

   Scrolling only occurs when `selectTarget` changes. `initialSelectIndex`
   initializes `tabSelectIndex` and the indicator offset but never emits
   an initial selection target. In an overflowing tab bar, an initial tab
   near the end can therefore remain off-screen.

5. **The shared scroll-event type fix is missing from the release plan**
   (`.changeset/oss-tabs-open.md:2-4`)

   `BaseScrollEvents` changes a public type contract, but the changesets
   still do not cover the affected packages. Please add patch changesets
   for:

   - `@lynx-js/lynx-ui-common`
   - `@lynx-js/lynx-ui-list`
   - `@lynx-js/lynx-ui-scroll-view`
   - `@lynx-js/lynx-ui-feed-list`
   - `@lynx-js/lynx-ui-input`

   Please also regenerate their API references.

**Review by MoonfaceX — APPROVED** [review]
